examples/locks_lib_example/Makefile

```diff
@@ -1,12 +1,12 @@
 # The TARGET variable determines what target system the application is
 # compiled for. It either refers to an XN file in the source directories
 # or a valid argument for the --target option when compiling
-TARGET = XK-1A
+TARGET ?= XK-1A
 
 # The APP_NAME variable determines the name of the final .xe file. It should
 # not include the .xe postfix. If left blank the name will default to
 # the project name
-APP_NAME = locks_lib_example
+APP_NAME = locks_lib_example_$(TARGET)
```

Comment:
What's the reason for not using the xcommon configs? You've pretty much recreated it yourself by appending the TARGET.

Reply:
Daniel explained this xcommon + xmake witchcraft to me and I will now implement it

Reply:
going to leave the tests the way they are though because they also have SW vs. HW config options and the way I have it at the moment works quite nicely

Reply:
> Daniel explained this xcommon + xmake witchcraft to me and I will now implement it

Please document anything you learn on a Confluence page.  Very few people know how to use xcommon and xmake.  The more we can learn about it, the better.